vm/safepoints.cpp

```diff
@@ -39,6 +39,15 @@ void safepoint_state::handle_safepoint(factor_vm* parent, cell pc) volatile {
     if (atomic::load(&parent->sampling_profiler_p))
       parent->end_sampling_profiler();
     std::cout << "Interrupted\n";
+    if (atomic::load(&parent->skip_debugger_p)) {
+      // Ctrl-Break throws an exception, interrupting the main thread, same
+      // as the "t" command in the factorbug debugger. But for Ctrl-Break to
+      // work we don't require the debugger to be activated, or even enabled.
+      atomic::store(&fep_p, false);
+      atomic::store(&parent->skip_debugger_p, false);
+      parent->general_error(ERROR_INTERRUPT, false_object, false_object);
+      FACTOR_ASSERT(false);
+    }
     parent->factorbug();
     atomic::store(&fep_p, false);
   } else if (atomic::load(&parent->sampling_profiler_p)) {
```

Comment:
I added it because of this code:
```
void safepoint_state::enqueue_fep(factor_vm* parent) volatile {
  if (parent->fep_p)
    fatal_error("Low-level debugger interrupted", 0);
```
I don't want my code to cause an interruption of the debugger, it's only intended for user-code.